tests/e2e/multi_vc.go

```diff
@@ -0,0 +1,1827 @@
+
+	Steps:
+	1. Use the partially matching storage policy and create PVC
+	2. PVC should not go to bound, appropriate error should be shown
+	3. Perform cleanup
+	*/
+
+	ginkgo.It("PVC creation failure when wrong storage policy name is specified in SC", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		storagePolicyName := "shared-ds-polic"
+		scParameters[scParamStoragePolicyName] = storagePolicyName
+
+		storageclass, pvclaim, err := createPVCAndStorageClass(client,
+			namespace, nil, scParameters, "", nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to fail as invalid storage policy is specified in Storage Class")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client,
+			pvclaim.Namespace, pvclaim.Name, framework.Poll, time.Minute/2)
+		gomega.Expect(err).To(gomega.HaveOccurred())
+		framework.Logf("Volume Provisioning Failed for PVC %s due to invalid storage "+
+			"policy given in the Storage Class", pvclaim.Name)
+	})
+
+	/*
+		TESTCASE-13
+		Deploy workload With allowed topology of VC1 and datastore url which is in VC2
+
+		Steps:
+		1. Create SC with allowed topology which matches VC1 details and datastore url which is in VC2
+		2. PVC should not go to bound, appropriate error should be shown
+	*/
+
+	ginkgo.It("Deploy workload with allowed topology of VC1 and datastore url of VC2", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		topValStartIndex = 0
+		topValEndIndex = 1
+		scParameters[scParamDatastoreURL] = datastoreURLVC2
+
+		ginkgo.By("Set specific allowed topology")
+		allowedTopologies = setSpecificAllowedTopology(allowedTopologies, topkeyStartIndex, topValStartIndex,
+			topValEndIndex)
+
+		storageclass, pvclaim, err := createPVCAndStorageClass(client,
+			namespace, nil, scParameters, "", allowedTopologies, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to fail as non-compatible datastore url is specified in Storage Class")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client,
+			pvclaim.Namespace, pvclaim.Name, framework.Poll, time.Minute/2)
+		gomega.Expect(err).To(gomega.HaveOccurred())
+		framework.Logf("Volume Provisioning Failed for PVC %s due to non-compatible datastore "+
```

Comment:
Ack, fixed

Reply:
One note - I took it like this because the expected error message gets changed i.e. events get changed in each release, so we need to fix it in a new PR